src/double.ts

```diff
@@ -48,18 +48,24 @@ export class Double extends BSONValue {
    */
   static fromString(value: string): Double {
     const coercedValue = Number(value);
-    const nonFiniteValidInputs = ['Infinity', '-Infinity', 'NaN'];
+
+    if (value === 'NaN') {
+      return new Double(NaN);
+    } else if (value === 'Infinity') {
+      return new Double(Infinity);
+    } else if (value === '-Infinity') {
+      return new Double(-Infinity);
+    }
```

Comment:
refactored some of the function for easier code parsing and better error messages